### PR TITLE
create category

### PR DESCRIPTION
**Pull request by PhamVanTung**

### CODE REVIEW

File: app/controllers/admin/categories_controller.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+class Admin::CategoriesController < ApplicationController
+  before_action :require_admin, :user_signed_in?
```
Comment (nguyenduyta) [inline app/controllers/admin/categories_controller.rb]: exchange :user_signed_in? with authenticate_user
Search google for more info

File: app/controllers/categories_controller.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class CategoriesController < ApplicationController
+  before_action :user_signed_in?
```
Comment (nguyenduyta) [inline app/controllers/categories_controller.rb]: nt

File: app/views/admin/categories/_category.html.erb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<li class="category-<%= category.id %>">
+  <div class="row">
+    <span class="col-md-9">
+      <div class="cateName"><%= category.name %> </div>
```
Comment (nguyenduyta) [inline app/views/admin/categories/_category.html.erb]: remove space

File: app/controllers/admin/categories_controller.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
+class Admin::CategoriesController < ApplicationController
+  before_action :require_admin, :authenticate_user!
+  def index
+    @categories = Category.all
```
Comment (dieunb) [inline app/controllers/admin/categories_controller.rb]: pagination

File: app/controllers/categories_controller.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+class CategoriesController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @categories = Category.all
```
Comment (dieunb) [inline app/controllers/categories_controller.rb]: pagination

File: app/views/static_pages/home.html.erb [hunk 6]
```diff
@@ -1,11 +1,15 @@
-<div class="center jumbotron">
-  <h1>Welcome to the Test App</h1>
+<% if user_signed_in? %>
+	<h1>Home page</h1>
```
Comment (dieunb) [inline app/views/static_pages/home.html.erb]: indent

File: app/views/static_pages/home.html.erb [hunk 7]
```diff
@@ -1,11 +1,15 @@
...
+<% else %>
+	<div class="center jumbotron">
```
Comment (dieunb) [inline app/views/static_pages/home.html.erb]: indent

File: app/views/static_pages/home.html.erb [hunk 8]
```diff
@@ -1,11 +1,15 @@
...
+	  <h2>
+	    This is the home page for the
+	    <a href="http://www.railstutorial.org/">Ruby on Rails Tutorial</a>
+	    sample application.
+	  </h2>
 
-  <%= link_to "Sign up now!", new_user_registration_path, class: "btn btn-lg btn-primary" %>
-</div>
+	  <%= link_to "Sign up now!", new_user_registration_path, class: "btn btn-lg btn-primary" %>
+	</div>
```
Comment (dieunb) [inline app/views/static_pages/home.html.erb]: indent

**Pull request merged**